lib/Sema/ConstraintSystem.cpp

```diff
@@ -3002,6 +3017,20 @@ void ConstraintSystem::bindOverloadType(
       return;
     }
 
+    // The opened type of an unbound member reference has optionality applied
+    // to the uncurried type.
+    if (!doesMemberRefApplyCurriedSelf(choice.getBaseType(),
+                                       choice.getDecl())) {
+      bindTypeOrIUO(openedType);
+      return;
+    }
+
+    if (choice.getDecl()->isInstanceMember() &&
+        choice.getBaseType()->is<AnyMetatypeType>()) {
+      bindTypeOrIUO(openedType);
+      return;
+    }
+
```

Comment:
Note to self: retrospectively it looks like checking for a metatype base is enough here.